mlflow/data/evaluation_dataset.py

```diff
@@ -130,11 +172,17 @@ def _hash_array_like_element_as_bytes(v):
             except TypeError:
                 return b""  # Skip unhashable types by returning an empty byte string
 
-        if Version(pd.__version__) >= Version("2.1.0"):
-            data = data.map(_hash_array_like_element_as_bytes)
+        if is_polars_dataframe(data):
+            return _hash_polars_table_as_bytes(data)
+        elif is_pyarrow_table(data):
+            return _hash_pyarrow_table_as_bytes(data)
```

Comment:
I don't love having these specific paths but I couldn't manage to adapt the workflow to be generic. It is the only instance in which I had to introduce a dedicated polars and pyarrow path